src/baggage/remote_baggage_restriction_manager.js

```diff
@@ -0,0 +1,161 @@
+// @flow
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Metrics from '../metrics/metrics.js';
+import NullLogger from '../logger.js';
+import NoopMetricFactory from '../metrics/noop/metric_factory';
+import * as constants from './default_baggage_restriction_manager';
+import Utils from '../util.js';
+
+const DEFAULT_REFRESH_INTERVAL = 60000;
+const DEFAULT_BAGGAGE_HOST = '0.0.0.0';
+const DEFAULT_BAGGAGE_PORT = 5778;
+const DEFAULT_FAIL_CLOSED = false;
+
+export default class RemoteBaggageRestrictionManager {
+
+    _serviceName: string;
+    _logger: Logger;
+    _metrics: Metrics;
+
+    _refreshInterval: number;
+    _host: string;
+    _port: number;
+    _failClosed: boolean;
+    _restrictionsMap: { [key: string]: number };
+
+    _initialDelayTimeoutHandle: any;
+    _refreshIntervalHandle: any;
+    _onRestrictionsUpdate: ?Function;
+
+    _initialized: boolean;
+
+    /**
+     * Creates a BaggageRestrictionManager that fetches baggage restrictions remotely from jaeger-agent.
+     *
+     * @param {string} [serviceName] - name of the current service / application, same as given to Tracer
+     * @param {object} [options] - optional settings
+     * @param {object} [options.logger] - optional logger, see _flow/logger.js
+     * @param {object} [options.failClosed] - this determines the startup failure mode of RemoteBaggageRestrictionManager.
```

Comment:
I thought so too, but according to Yuri: "it's a very common engineering term and you should be ashamed for not knowing it"